testnet/stacks-node/src/neon_node.rs

```diff
@@ -1360,73 +1361,58 @@ impl BlockMinerThread {
     fn make_vrf_proof(&mut self) -> Option<VRFProof> {
         // if we're a mock miner, then make sure that the keychain has a keypair for the mocked VRF
         // key
-        if self.config.node.mock_mining {
-            self.keychain.rotate_vrf_keypair(VRF_MOCK_MINER_KEY);
-        }
-
-        // Generates a proof out of the sortition hash provided in the params.
-        let vrf_proof = match self.keychain.generate_proof(
-            &self.registered_key.vrf_public_key,
-            self.burn_block.sortition_hash.as_bytes(),
-        ) {
-            Some(vrfp) => vrfp,
-            None => {
-                // Try to recover a key registered in a former session.
-                // registered_key.block_height gives us a pointer to the height of the block
-                // holding the key register op, but the VRF was derived using the height of one
-                // of the parents blocks.
-                let _ = self
-                    .keychain
-                    .rotate_vrf_keypair(self.registered_key.block_height - 1);
-                match self.keychain.generate_proof(
-                    &self.registered_key.vrf_public_key,
-                    self.burn_block.sortition_hash.as_bytes(),
-                ) {
-                    Some(vrfp) => vrfp,
-                    None => {
-                        error!(
-                            "Relayer: Failed to generate proof with {:?}",
-                            &self.registered_key.vrf_public_key
-                        );
-                        return None;
-                    }
-                }
-            }
+        let vrf_proof = if self.config.node.mock_mining {
+            self.keychain.generate_proof(
+                VRF_MOCK_MINER_KEY,
+                self.burn_block.sortition_hash.as_bytes(),
+            )
+        } else {
+            self.keychain.generate_proof(
+                self.registered_key.target_block_height,
+                self.burn_block.sortition_hash.as_bytes(),
+            )
         };
 
         debug!(
-            "Generated VRF Proof: {} over {} with key {}",
+            "Generated VRF Proof: {} over {} ({},{}) with key {}",
             vrf_proof.to_hex(),
             &self.burn_block.sortition_hash,
+            &self.burn_block.block_height,
+            &self.burn_block.burn_header_hash,
             &self.registered_key.vrf_public_key.to_hex()
         );
         Some(vrf_proof)
     }
 
     /// Get the microblock private key we'll be using for this tenure, should we win.
-    /// Return the private key on success
-    /// return None if we were unable to generate the key.
-    fn make_microblock_private_key(&mut self, attempt: u64) -> Option<Secp256k1PrivateKey> {
+    /// Return the private key.
+    ///
+    /// In testing, we ignore the parent stacks block hash because we don't have an easy way to
+    /// reproduce it in integration tests.
+    #[cfg(not(test))]
```

Comment:
Sorry, I'm still stuck on this. I see you swapped the attributes, but I think the comment should now be moved to the other function. Also, the parameter `_parent_stacks_hash` is still not used in this version either.

Reply:
Sorry, the rules should be fixed now as follows:

* In test mode, use the block height's big-endian representation.

* In prod, use the parent index block hash.